Bottom: 9ea9ebd1032f8ba44641e7b4b8c324743f7917aa
Top:    20d0bff459590313535e98a0130a647140ecec97
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-24 17:22:39 -0700

Refresh of rs-opt-inline

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 7107328..7802ef5 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -105,6 +105,7 @@ struct rs_sge {
 	uint32_t length;
 };
 
+#define RS_INLINE_MIN    (sizeof(struct rs_sge))
 #define rs_host_is_net() (1 == htonl(1))
 #define RS_CONN_FLAG_NET 1
 
@@ -1715,6 +1716,8 @@ int rsetsockopt(int socket, int level, int optname,
 			break;
 		case RDMA_INLINE:
 			rs->sq_inline = min(*(uint32_t *) optval, RS_QP_MAX_SIZE);
+			if (rs->sq_inline < RS_INLINE_MIN)
+				rs->sq_inline = RS_INLINE_MIN;
 			break;
 		default:
 			break;
